### ISSUE TITLE
when I train with tensorflow, an error about the confusion_matrix appeared

### ISSUE THREAD

**Issue by UnBuen** [klicperajo/ppnp#13]
Dear klicperajo, 
when I train with tensorflow,  if I choose save_result when I train, it will appear name 'conf_mat' is not defined. so can you help me to deal with this problem, thank you very much.
the problem is in ppnp/tensorflow/training.py line 129

**Comment by gasteigerjo** [comment]
It looks like you're the first person to use save_results with TensorFlow since I broke it in February 2019. 😆 

Fixed in 394425a9eeffbd95c09d68d6f0e1473ac9c2fd4a.

**Comment by UnBuen** [comment]
Dear klicperajo,
when I train with tensorflow, if I choose the fourth dataset(Ms Academic, the others is working) and train, it will appear "Cannot take a larger sample than population when 'replace=False' ". 
So can you help me to deal with this problem, thank you very much.
the problem is in ppnp/tensorflow/training.py line 35(maybe 34).
btw, if you have time, can you explain the meaning of ntrain_per_class and nstopping.
`idx_split_args = {'ntrain_per_class': 20, 'nstopping': 500, 'nknown': 1500, 'seed': 2413340114}`😆

**Comment by gasteigerjo** [comment]
For MS Academic you need to use a lower number of nodes in your development/known set. See `https://github.com/klicperajo/ppnp/blob/master/reproduce_results.ipynb`.

`ntrain_per_class` is the number of nodes per class in your training set.
`nstopping` is the number of nodes in the early stopping set.
See Figure 7 in the paper for an illustration of these sets.

**Comment by UnBuen** [comment]
> For MS Academic you need to use a lower number of nodes in your development/known set. See `https://github.com/klicperajo/ppnp/blob/master/reproduce_results.ipynb`.
> 
> `ntrain_per_class` is the number of nodes per class in your training set.
> `nstopping` is the number of nodes in the early stopping set.
> See Figure 7 in the paper for an illustration of these sets.

It's very helpful for me. Thank you, I'm sorry I didn't read the paper carefully.😆
